dotnet/src/SemanticKernel.UnitTests/AI/FunctionChoiceBehaviors/FunctionChoiceBehaviorTests.cs

```diff
@@ -0,0 +1,255 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Linq;
+using Microsoft.SemanticKernel;
+using Xunit;
+
+namespace SemanticKernel.UnitTests.Functions;
+
+/// <summary>
+/// Unit tests for <see cref="FunctionChoiceBehavior"/>
+/// </summary>
+public sealed class FunctionChoiceBehaviorTests
+{
+    private readonly Kernel _kernel;
+
+    public FunctionChoiceBehaviorTests()
+    {
+        this._kernel = new Kernel();
+    }
+
+    [Fact]
+    public void AutoFunctionChoiceShouldBeUsed()
+    {
+        // Act
+        var choiceBehavior = FunctionChoiceBehavior.Auto();
+
+        // Assert
+        Assert.IsType<AutoFunctionChoiceBehavior>(choiceBehavior);
+    }
+
+    //[Fact]
+    //public void RequiredFunctionChoiceShouldBeUsed()
+    //{
+    //    // Act
+    //    var choiceBehavior = FunctionChoiceBehavior.Required();
+
+    //    // Assert
+    //    Assert.IsType<RequiredFunctionChoiceBehavior>(choiceBehavior);
+    //}
+
+    //[Fact]
+    //public void NoneFunctionChoiceShouldBeUsed()
+    //{
+    //    // Act
+    //    var choiceBehavior = FunctionChoiceBehavior.None();
+
+    //    // Assert
+    //    Assert.IsType<NoneFunctionChoiceBehavior>(choiceBehavior);
+    //}
```

Comment:
Comments or Skip?

Reply:
They will be needed later to test other parts of the new function calling model when they are added one after another. It'll be challenging to keep track of tests that were copied from the old big function calling abstraction PR and which were not. So, I'll keep them commented out for a while.